infrastructure/simulation-tool/src/utils/number.utils.ts

```diff
@@ -0,0 +1,24 @@
+import { BigNumber, BigNumberish } from 'ethers';
+
+type RandomAmountGenerator = Generator<BigNumber>;
+
+const getRandomBigNumber = (min: BigNumberish, max: BigNumberish): BigNumber => {
+    const diff = BigNumber.from(1).mul(max).sub(min);
+    const randomDecimal = Math.random();
+    const scaled = diff.mul(BigNumber.from((randomDecimal * 1e18).toFixed()));
+    const randomNumber = scaled.div(BigNumber.from((1e18).toString()));
+
+    return randomNumber.add(min);
+};
+
+function* createRandomAmountGenerator(minMaxTuple: [BigNumberish, BigNumberish]): Generator<BigNumber> {
```

Comment:
I don't see the purpose of having a generator when we can just call the function directly and have the same result.